libnavui-base/src/main/java/com/mapbox/navigation/ui/base/installer/ComponentInstaller.kt

```diff
@@ -104,12 +104,11 @@ fun MapboxNavigationApp.installComponents(
  * ```
  */
 @ExperimentalPreviewMapboxNavigationAPI
-fun MapboxNavigation.installComponents(
-    lifecycleOwner: LifecycleOwner,
+fun LifecycleOwner.installComponents(
```

Comment:
Please revert this. This change breaks desired public API.
The `MapboxNavigation.installComponents` extension allows component installation in a receiving `MapboxNavigation` instance, when passed `LifecycleOwner` enters `ON_CREATED` state. Furthermore, by extending `MapboxNavigation` we ensure that developers create a valid MapboxNavigation instance before calling installing components.

Reply:
`MapboxNavigation` instances are received through `MapboxNavigationObserver` 🤔 

This is fixing an issue where installed components will ignore `MapboxNavigation` changes. @tomaszrybakiewicz will you provide reproduce steps for the issue you're seeing? because it's working for me

Reply:
I understand that MapboxNavigaiton can be acquired through `MapboxNavigationObserver` interfaces. However, the purpose of this API is to enforce the correct order of operations: **developers must create a valid MapboxNavigation instance before installing components**.

Since MapboxNavigaiton instance can be created in many ways, the ComponentsInstaller API offers two mechanisms for installing components.

1. Via MapboxNavigationApp Facade.
  This is useful when the MapboxNavigation instance is initialized either outside Fragment or Activity (eg. in their Application class) or when its initialization has to be deferred till later.
    ```kotlin
    MapboxNavigationApp.installComponents(/*lifecycleOwner*/ this) {
        audioGuidanceButton(binding.soundButton)
        routeLine(binding.mapView)
        routeArrow(binding.mapView)
    }
    ```
2. Via existing MapboxNavigation instance.
  This is useful when developers already have access to MapboxNavigation instance and just want to install components.
    ```kotlin
    mapboxNavigation.installComponents(/*lifecycleOwner*/ this) {
        audioGuidanceButton(binding.soundButton)
        routeLine(binding.mapView)
        routeArrow(binding.mapView)
    }
    ```

Reply:
>  developers must create a valid MapboxNavigation instance before installing components.

The purpose of `MapboxNavigationApp` is to simplify this part of installing. Otherwise, developers need to rebuild the internals `MapboxNavigationApp`

Anything that uses `mapboxNavigation.installComponents` will break when they change `MapboxNavigationApp.setup`

A separate point, is maybe the better interface is `MapboxNavigationApp.installComponents(/*lifecycleOwner*/ this)` instead of `LifecycleOwner.installComponents` 🤔

Reply:
Am removing the `LifecycleOwner.installComponents` in favor of `MapboxNavigationApp.installComponents`

Reply:
Kyle I agree with you that we should probably remove `mapboxNavigation.installComponents`. However, as we discussed I think the timing is not yet right for this removal. The reason being `MapboxNavigationApp` is not yet enforced to be used to create `mapboxNavigation`. So if users are using `MapboxNavigationProvider` to create `mapboxNavigation` how can we be sure that `mapboxNavigation` exists before calling `LifecycleOwner.installComponents`. I think we just need to wait for `MapboxNavigationApp` to become the only way you create `mapboxNavigation` and then we can use `LifecycleOwner.installComponents`

Reply:
> I think we just need to wait for MapboxNavigationApp to become the only way you create mapboxNavigation and then we can use LifecycleOwner.installComponents

But `installComponents` is a new function that is not adopted yet 🤔. Why are we in favor of experimental APIs that we know we want to deprecate?

Reply:
> So if users are using MapboxNavigationProvider to create mapboxNavigation how can we be sure that mapboxNavigation exists before calling LifecycleOwner.installComponents. I

Plan is to deprecate `MapboxNavigationProvider` so again, super confusing why we want to support APIs that are not what we want

This pull request was a small suggestion and it received a lot of friction. Am closing